aten/src/ATen/native/cuda/SummaryOps.cu

```diff
@@ -260,16 +272,53 @@ Tensor _bincount_cuda_template(
 
   auto nbins = self.max().item<int64_t>() + 1L;
   nbins = std::max(nbins, minlength);
+  input_t minvalue = 0;
+  input_t maxvalue = *self.max().cpu().data<input_t>();
+  input_t binsize = 1;
   // alloc output counter on GPU
   Tensor output;
   if (has_weights) {
     output = native::zeros({nbins}, weights.options());
-    auto ret = cuda::CUDA_tensor_histogram<weights_t, input_t, true>(
-        output, self, weights, nbins, 1);
+    auto ret = cuda::CUDA_tensor_histogram<weights_t, input_t, input_t, true>(
+        output, self, weights, nbins, minvalue, maxvalue, binsize);
   } else {
     output = native::zeros({nbins}, device(DeviceType::CUDA).dtype(kLong));
-    auto ret = cuda::CUDA_tensor_histogram<int64_t, input_t, false>(
-        output, self, weights, nbins, 1);
+    auto ret = cuda::CUDA_tensor_histogram<int64_t, input_t, input_t, false>(
+        output, self, weights, nbins, minvalue, maxvalue, binsize);
+  }
+  return output;
+}
+
+///////////////// histc /////////////////
+template <typename input_t>
+Tensor _histc_cuda_template(
+    const Tensor& self,
+    int64_t nbins,
+    input_t min,
+    input_t max) {
+  //FIXME histc_cpu has no boundary check, do they need to be consistent?
+  if (nbins < 0) {
+    AT_ERROR("nbins should be > 0");
+  }
+  Tensor output = native::zeros({nbins}, device(DeviceType::CUDA).dtype(kLong));
+  input_t minvalue;
+  input_t maxvalue;
+  if (min == max) {
+    minvalue = *self.min().cpu().data<input_t>();
+    maxvalue = *self.max().cpu().data<input_t>();
+  }
+  if (minvalue == maxvalue) {
+    minvalue = minvalue - 1;
+    maxvalue = maxvalue + 1;
+  }
+  auto binsize = (maxvalue - minvalue) / nbins;
+  if (binsize == ::floor(binsize)) {
+    auto ret = cuda::CUDA_tensor_histogram<int64_t, input_t, int64_t, false>(
+      output, self, at::empty({0}, self.options()), nbins, minvalue, maxvalue, binsize);
+  } else {
+    //FIXME is this okay? either int or double
```

Comment:
What is the question here?

Reply:
I don't think it makes sense to always make it int, even if the inputs are of type int the binsize might be fractional. Always float or double should be the way to go.

Reply:
oh yea, `int` is def not sufficient. what I wasn't sure is whether we need to support both float and double for fractional binsize. but you're right, `int` is not even needed, I'll make it either float or double. thanks!